test/tests/functional/pbs_nodes_json.py

```diff
@@ -91,3 +91,33 @@ def test_check_tab_escape_json(self):
             json.loads("\n".join(n_out))
         except ValueError:
             self.assertFalse(True, "Json failed to load")
+
+    @tags('smoke')
+    def test_validate_pbsnodes_attrs(self):
+        """
+        Test if the pbsnodes outputs all the default attributes.
+        And does not print any new attributes.
```

Comment:
You shouldn't check for an exact set of attributes.  In the future if we  add a new default attribute, your test will fail.  Since you are tagging this test as smoke, it will be run by travis.  This will hold up any subsequent PR until we fix it.  Fixing it requires Subhasis to jump through hoops.

Reply:
The intention is to check for the default set of attributes of pbsnodes output. Nobody should add/remove an attribute to/from pbsnodes output mistakenly. If they do, this test case will trigger a red flag on travis and the PR will be blocked. If the change is intentional, the smoke test also should get updated along with the PR.

Reply:
If that is what you want to do, then I'm OK with it.  I'm not sure the added maintenance of checking for things that shouldn't show up is warranted, but I see the point of it.

Reply:
@arungrover and I also don't understand the benefit of adding the smoke test.
We're not sure we understand how someone "mistakenly" adds/removes an attribute.  Can you please explain more?  It may help us to understand why such a smoke test is needed.
Because in my opinion if someone does mistakenly add/remove and attribute and the smoke test fails then the developer will see the red flag and change the smoke test.  No one else will necessarily be aware of any mistaken add/remove because the tests will now pass.
It only seems to add maintenance as @bhroam already mentioned.

Reply:
It is helpful in catching bugs such as additional pbsnodes attribute `license='u'` which had addressed with this PR. In scenarios like these, this test will fail and notifies the developer that there is something strange. The test description explains the motives behind it. Even if the developer doesn't observe it and updates the test case to clear the red flag, the maintainer can raise the concern on why a smoke test got edited without appropriate explanation/design document.

This test also doesn't add any bottleneck to travis CI as it does not submit a new job, and will get completed quickly. I cannot add this test case without a smoke tag as it will not help in achieving the objective. I would say the maintenance cost of this test is not significant as the features which modifies the node attributes are not frequent and that can be justified by preventing possible bugs.

Reply:
Thank you for the explanation @nithinj.  I suppose it's fine.   We can always remove it later if it turns out to hurt more than help. Thanks again.